api/metadata.proto

```diff
@@ -78,12 +78,16 @@ message RxMetadata {
   bytes uplink_token = 15;
   // Index of the gateway channel that received the message.
   uint32 channel_index = 17 [(validate.rules).uint32 = {lte: 255}];
+  // Hopping width (grid) for LR-FHSS in kHz.
+  string hopping_width = 19;
```

Comment:
The spec says:

```
 hpw  | number | LR-FHSS hopping channel width (grid number of steps)
```

So it's a number (not string) and the unit is steps (not KHz). Also, if it were KHz, we would be using Hz as everywhere in our API.

So where does this come from?

Also can you add a test case for unmarshaling a message?

Reply:
Ah yeah right.
```
 hpw  | string | LR-FHSS hopping channel width (grid), in kHz
```
That's from the old spec file [`packet_forwarder_PROTOCOL.txt`](https://github.com/TheThingsNetwork/lorawan-stack/issues/3806#issuecomment-823093084)  in the issue.

This has indeed changed in the new spec.